enzyme/Enzyme/TypeAnalysis/TypeTree.h

```diff
@@ -248,22 +251,63 @@ class TypeTree : public std::enable_shared_from_this<TypeTree> {
 
         for (const auto &val : toremove) {
           mapping.erase(val);
+          changed = true;
         }
       }
     }
     if (Seq.size() > 6) {
       llvm::errs() << "not handling more than 6 pointer lookups deep dt:"
                    << str() << " adding v: " << to_string(Seq) << ": "
                    << CT.str() << "\n";
-      return;
+      return changed;
+    }
+
+    std::vector<std::pair<int, std::set<std::vector<int>>>> best;
+    for (const auto &pair : mapping) {
+      size_t i = 0;
+      for (int val : pair.first) {
+        if (best.size() <= i) {
+          best.emplace_back(val, std::set<std::vector<int>>());
+        }
+        assert(best.size() > i);
+        if (val > best[i].first) {
+          best[i].first = val;
+          best[i].second.clear();
+        }
+
+        if (val == best[i].first) {
+          best[i].first = val;
+          best[i].second.insert(pair.first);
+        }
+        i++;
+      }
     }
+    size_t i = 0;
+    bool keep = false;
+    bool considerErase = false;
     for (auto Off : Seq) {
-      if (Off > 500) {
-        // TODO perhaps issue warning for too large an offset
-        return;
+      if (i < best.size()) {
+        if (Off < best[i].first) {
+          if (best[i].first > 500)
```

Comment:
turn this into an option? Or at least document why `500` was choosen as a cutoff